src/charm.py

```diff
@@ -289,6 +317,36 @@ def _install_arguments(self) -> List[str]:
             arguments.extend([f"--{key}", value])
         return arguments
 
+    @property
+    def _magma_service_is_running(self) -> bool:
+        """Checks whether magma is running."""
+        magma_service = subprocess.run(
+            ["systemctl", "is-active", "magma@magmad"],
+            stdout=subprocess.PIPE,
+        )
+        if magma_service.returncode != 0:
+            return False
+        return True
+
+    @property
+    def _get_magma_secrets(self) -> Tuple[Optional[str], Optional[str]]:
+        """Gets Access Gateway's Hardware ID and Challenge key.
+
+        Hardware ID and Challenge key are available through the `show_gateway_info` script
+        provided by the Access Gateway. This method filers out required values from the script's
+        output.
+
+        Returns:
+            str: Hardware ID
+            str: Challenge key
+        """
+        gateway_info = subprocess.check_output(["show_gateway_info.py"]).decode().split("\n")
+        gateway_info = list(filter(None, gateway_info))
+        gateway_info = list(filter(lambda x: (not re.search("^-(-*)", x)), gateway_info))
+        hardware_id = gateway_info[gateway_info.index(self.HARDWARE_ID_LABEL) + 1]
```

Comment:
How feasible would it be to use the same logic as in AGW to get the hardware_id and challenge_key (even if it means some code duplication) instead of looking at printed output? I'm asking because I imagine any change in the output format could cause this action to break.

Reply:
AGW snap simply re-prints the output of the `show_gateway_info.py`. I agree that current implementation brings the risk of additional maintenance, but then again I don't expect the output to change too often if at all. Also re-printing the whole output would look really ugly ;)

Reply:
I don't think we need to print the whole output.
```suggestion
import snowflake
from magma.common.cert_utils import load_public_key_to_base64der
public_key = load_public_key_to_base64der(opts.pub_key)
challenge_key = public_key.decode('utf-8')
hardware_id = snowflake.snowflake()
```
But I'm not sure how would we be able to access `magma.common.cert_utils`, if that's what you mean by "ugly", then I'm okay with how it is right now :)

Reply:
We wouldn't. I've already tried that while working on the snap. All dependencies need to be duplicated. 
Also I think that the code responsible for generating the secrets is actually more likely to change than the script output format.

Reply:
Got it :) looks good to me in that case :)